taqueria-vscode-extension/src/lib/helpers.ts

```diff
@@ -61,45 +63,57 @@ export const sanitizeDeps = (deps?: InjectedDependencies) =>
 export const inject = (deps: InjectedDependencies) => {
 	const { vscode } = deps;
 
-	const exposeInitTask = async (
+	const exposeInitTask = (
 		context: api.ExtensionContext,
 		output: Output,
 		i18n: i18n,
 		folders: readonly api.WorkspaceFolder[],
 	) => {
-		const exposeTask = exposeTaqTaskAsCommand(context, output, i18n);
-
-		// As the developer has no folder open, we must prompt
-		// them for a path they would like to taqify
-		if (folders.length === 0) {
-			addCommand(context)(Commands.init, () => {
-				return vscode.window.showOpenDialog({
-					canSelectFolders: true,
-					canSelectFiles: false,
-					openLabel: 'Select project folder',
-					title: "Select a project folder to taq'ify",
-					canSelectMany: false,
-				})
-					.then(uris =>
-						uris?.map(
-							uri =>
-								getTaqBinPath(i18n)
-									.then(pathToTaq => Util.proxyToTaq(pathToTaq, i18n, undefined)(`init ${uri.path}`))
-									.then(_ => vscode.window.showInformationMessage("Project taq'fied!", uri.path))
-									.then(_ => vscode.workspace.updateWorkspaceFolders(0, undefined, { uri })),
-						)
-					)
-					.then(console.log) as Promise<void>;
-			});
-		} // The developer has one folder open. We can assume that
-		// is the folder they wish to taqify
-		else if (folders.length === 1) {
-			return exposeTask('taqueria.init', `init ${folders[0].uri.path}`, 'notify');
-		} // The developer has multiple folders in the their workspace.
-		// As can't really know which one to taqify without prompting them
-		else {
-			console.log('Coming soon!');
-		}
+		addCommand(context)(Commands.init, async () => {
+			const uri = await getFolderForScaffolding(context, output, i18n, folders);
+			if (uri === undefined) {
+				return;
+			}
+			getTaqBinPath(i18n)
+				.then(pathToTaq => Util.proxyToTaq(pathToTaq, i18n, undefined)(`init ${uri.path}`))
+				.then(_ => vscode.window.showInformationMessage("Project taq'fied!", uri.path))
+				.then(_ => vscode.workspace.updateWorkspaceFolders(0, undefined, { uri }))
+				.then(console.log);
```

Comment:
How will errors be handled? There's no `.catch` for this promise.

Reply:
It didn't, nice catch Jimmy.